language/move-vm/runtime/src/data_cache.rs

```diff
@@ -1,33 +1,42 @@
 // Copyright (c) The Libra Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 
-use libra_types::{
-    access_path::AccessPath,
-    contract_event::ContractEvent,
-    on_chain_config::ConfigStorage,
+use crate::loader::Loader;
+
+use move_core_types::{
+    account_address::AccountAddress,
+    identifier::Identifier,
+    language_storage::{ModuleId, TypeTag},
+    value::MoveTypeLayout,
     vm_status::{StatusCode, VMStatus},
-    write_set::{WriteOp, WriteSet, WriteSetMut},
 };
-use move_core_types::language_storage::ModuleId;
 use move_vm_types::{
     data_store::DataStore,
-    loaded_data::types::FatStructType,
-    values::{GlobalValue, Struct, Value},
+    loaded_data::runtime_types::Type,
+    values::{GlobalValue, Value},
 };
-use std::{collections::btree_map::BTreeMap, mem::replace};
-use vm::errors::*;
+use std::collections::btree_map::BTreeMap;
+use vm::{errors::*, file_format::CompiledModule};
 
 /// Trait for the Move VM to abstract `StateView` operations.
 ///
 /// Can be used to define a "fake" implementation of the remote cache.
 pub trait RemoteCache {
-    fn get(&self, access_path: &AccessPath) -> VMResult<Option<Vec<u8>>>;
+    fn get_module(&self, module_id: &ModuleId) -> VMResult<Option<Vec<u8>>>;
+    fn get_resource(&self, address: &AccountAddress, tag: &TypeTag) -> VMResult<Option<Vec<u8>>>;
+}
+
+pub struct AccountDataCache {
+    data_map: BTreeMap<Type, Option<GlobalValue>>,
+    module_map: BTreeMap<Identifier, (Vec<u8>, CompiledModule)>,
```

Comment:
as per comment below, I am not sure `CompiledModule` is proper here, but I am not clear yet if removing it is an issue

Reply:
Good point. I'll think about this and put up a follow up PR if necessary.